src/new-client/testing/log-in.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { test, expect } from '@playwright/test';
+import env from './testEnvironment';
+
+const { TEST_USER_USERNAME, TEST_USER_PASSWORD } = env;
+
+test('Loads the site', async ({ page }) => {
+  await page.goto('/');
+
+  await expect(page).toHaveTitle('Appointment Management Service');
+});
+
+test('Has a log in link', async ({ page }) => {
+  await page.goto('/');
+
+  await expect(page.getByRole('button', { name: 'Log In' })).toBeVisible();
+});
+
+test('logs in', async ({ page }) => {
+  await page.goto('/');
+
+  await expect(page.getByRole('button', { name: 'Log In' })).toBeVisible();
+
+  await page.getByRole('button', { name: 'Log In' }).click();
+
+  await page.getByLabel('Username').fill(TEST_USER_USERNAME);
+  await page.getByLabel('Password').fill(TEST_USER_PASSWORD);
+
+  await page.getByLabel('Password').press('Enter');
+
+  await expect(
+    page.getByRole('heading', { name: 'Choose a site' }),
+  ).toBeVisible();
+});
+
+test('logs out', async ({ page }) => {
+  await page.goto('/');
+  await page.getByRole('button', { name: 'Log In' }).click();
+  await page.getByLabel('Username').fill(TEST_USER_USERNAME);
+  await page.getByLabel('Password').fill(TEST_USER_PASSWORD);
+  await page.getByLabel('Password').press('Enter');
+
+  await expect(
+    page.getByRole('heading', { name: 'Choose a site' }),
+  ).toBeVisible();
+
+  await expect(page.getByRole('button', { name: 'Log Out' })).toBeVisible();
+  await page.getByRole('button', { name: 'Log Out' }).click();
+
+  await expect(
+    page.getByRole('heading', { name: 'You cannot access this site' }),
+  ).toBeVisible();
+  await expect(
+    page.getByText(
+      'You are currently not signed in. To use this site, please sign in.',
+    ),
+  ).toBeVisible();
+});
+
+test('Creates a token cookie upon sign in', async ({ browser }) => {
```

Comment:
This is testing implementation. Usually E2E tests are testing outcomes - i.e. ensure the user can login and is shown the correct page etc.

Reply:
Yeah, I almost didn't add it for that reason but I just wanted to prove that Playwright was capable of interacting with the browser storage. I was imagining we could write tests to remove the cookie then do a page transition or something. It was silly, I'll remove it.

Reply:
As discussed on our call, have removed :)